src/App.tsx

```diff
@@ -1,31 +1,54 @@
-import useViewStore from './stores/view';
+import useViewStore, { getNavigationProps } from './stores/view';
 import Landing from './components/Landing';
 import Upload from './components/Upload';
 import ColumnAnnotation from './components/ColumnAnnotation';
 import ValueAnnotation from './components/ValueAnnotation';
 import Download from './components/Download';
+import NavigationButton from './components/NavigationButton';
+import StepperComponent from './components/Stepper';
+import { View } from './utils/types';
 
 function App() {
   const currentView = useViewStore((state) => state.currentView);
 
+  const { backView, nextView, backLabel, nextLabel, className } = getNavigationProps(currentView);
+
   const renderView = () => {
     switch (currentView) {
-      case 'landing':
+      case View.Landing:
         return <Landing />;
-      case 'upload':
+      case View.Upload:
         return <Upload />;
-      case 'columnAnnotation':
+      case View.ColumnAnnotation:
         return <ColumnAnnotation />;
-      case 'valueAnnotation':
+      case View.ValueAnnotation:
         return <ValueAnnotation />;
-      case 'download':
+      case View.Download:
         return <Download />;
       default:
         return <Landing />;
     }
   };
 
-  return <div>{renderView()}</div>;
+  return (
+    <div className="flex min-h-screen flex-col overflow-x-hidden">
+      {currentView !== View.Landing && <StepperComponent currentView={currentView} />}
+
+      {renderView()}
+
+      {currentView !== View.Landing && currentView !== View.Download && (
```

Comment:
Similarly here I would add a comment explaining this is the section for the bottom-sticky navigation elements